src/Elyon.Fastly.Api.DomainServices/InfoSessionFollowUpService.cs

```diff
@@ -49,18 +53,128 @@ public async Task SendInfoSessionFollowUpEmailAsync(InfoSessionFollowUpSpecDto s
+                        { "OnboardingDate", ((DateTime)organization.OnboardingTimestamp).ToString("d", CultureInfo.CreateSpecificCulture("de-CH")) },
+                        { "areaPharmacy", organization.Area },
+                        { "CompanyShortcut", organization.OrganizationShortcutName },
+                        { "firstTestingDate", ((DateTime)organization.FirstTestTimestamp).ToString("d", CultureInfo.CreateSpecificCulture("de-CH")) },
+                        { "TestingDay", ((DateTime)organization.FirstTestTimestamp).ToString("dddd", CultureInfo.CreateSpecificCulture("de-CH")) }
+                    };
+                    
+                    await _emailSenderService.SendOnboardingEmail(receiver, specDto.CcReceivers, organization.OrganizationTypeId, parameters).ConfigureAwait(false);
+                }
+            }
+            else if (organization.OrganizationType.Id == _companyOrganizationId)
+            {
+                if (!ValidateOrganizationForOnboarding(organization))
+                    return;
 
-            foreach (var receiver in specDto.Receivers)
+                foreach (var receiver in specDto.Receivers)
+                {
+                    var parameters = new Dictionary<string, string>
+                    {
+                        { "CompanyShortcut", organization.OrganizationShortcutName },
+                        { "supportPerson", organization.SupportPerson.Name }
+                    };
+                    await _emailSenderService.SendOnboardingEmail(receiver, specDto.CcReceivers, organization.OrganizationTypeId, parameters).ConfigureAwait(false);
+                }
+            }
+            else
             {
-                await _emailSenderService.SendInfoSessionFollowUpEmail(receiver, specDto.Message, generatedToken)
-                    .ConfigureAwait(false);
+                if (string.IsNullOrEmpty(specDto.Message))
+                {
+                    ValidationDictionary.AddModelError("Message", "Message must not be null or empty space");
+                    return;
+                }
+
+                if (!ValidateOrganizationForNewFollowUp(organization))
+                    return;
+
+                string generatedToken = await AddFollowUpAsync(organization).ConfigureAwait(false);
+
+                foreach (var receiver in specDto.Receivers)
+                {
+                    await _emailSenderService.SendInfoSessionFollowUpEmail(receiver, specDto.Message, generatedToken)
+                        .ConfigureAwait(false);
+                }
             }
         }
 
+        private bool ValidateOrganizationForOnboarding(OrganizationDto organization)
+        {
+            if (organization == default)
+            {
+                ValidationDictionary.AddModelError("OrganizationId", "Organization with specified Id does not exist");
```

Comment:
this error doesn't match the condition. It's not the id missing, it's the organization